.github/workflows/main-analysis.yml

```diff
@@ -3,15 +3,61 @@ on:
   push:
     branches: [ main ]
 jobs:
+  build-tauri:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ windows-latest ]
+    runs-on: ${{ matrix.platform }}
+    name: Build tauri app
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 18.4.0
+      - name: Install Rust stable
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - name: Install app dependencies
+        run: yarn install
+      - name: Build app
+        run: yarn build
+      - name: Run linter
+        run: yarn lint
+      - name: Run tests
+        run: yarn test a --coverage
+      - name: Upload coverage report
+        uses: actions/upload-artifact@master
+        with:
+          name: coverage
+          path: coverage/
+      - name: Build Tauri native binaries
+        uses: tauri-apps/tauri-action@v0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   sonarcloud:
     name: SonarCloud
+    if: ${{ success() }}
+    needs: build-tauri
```

Comment:
There's no need to specify `if: success` when using a `needs: previous_job`. It will [automatically stop execution if the needed job fails unless you specify no to do so](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds).